test/integ/test_cursor.py

```diff
@@ -975,6 +975,15 @@ def test_fetch_out_of_range_timestamp_value(conn, result_format):
             cur.fetchone()
 
 
+@pytest.mark.skipolddriver
+def test_null_in_non_null(conn):
+    with conn() as cnx:
+        cur = cnx.cursor()
+        cur.execute("create table null_in_non_null(bar char not null)")
```

Comment:
Please use our table name randomizer instead of using a hard-coded table name.

Also make this a temporary table, that way you won't need drop it in a finally clause.